### PR TITLE
feat: [ANDROAPP-3005] Carousel view for map display

### PR DESCRIPTION
**Pull request by Balcan**
## Description
[ jira issue ](https://jira.dhis2.org/browse/ANDROAPP-3005)

## Covered unit test cases
TBD
## Where did you test this issue?
- [x] Smartphone Emulator
- [x] Tablet Emulator
- [x] Smartphone
- [x] Tablet
## Which Android versions did you test this issue?
- [x] 4.4
- [ ] 5.X - 6.X
- [x] 7.X
- [x] 8.X
- [ ] 9.X - 10.X
- [ ] Other
## Checklist
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code

### CODE REVIEW

File: app/src/androidTest/java/org/dhis2/usescases/teidashboard/robot/TeiDashboardRobot.kt [hunk 1]
```diff
@@ -113,12 +113,24 @@ class TeiDashboardRobot : BaseRobot() {
...
-                                allOf(
-                            withId(R.id.event_name),
-                            withText(eventName))))))))
+            .check(
+                matches(
+                    allOf(
+                        isDisplayed(), isNotEmpty(),
+                        atPosition(
+                            0,
+                            hasDescendant(
```
Comment (JaimeToca) [inline app/src/androidTest/java/org/dhis2/usescases/teidashboard/robot/TeiDashboardRobot.kt:122]: please do not format this class

File: app/src/main/java/org/dhis2/uicomponents/map/geometry/line/MapLineRelationshipToFeature.kt [hunk 2]
```diff
@@ -27,4 +32,85 @@ class MapLineRelationshipToFeature {
...
+        return if (fromGeometry.type() == FeatureType.POINT &&
+            toGeometry.type() == FeatureType.POINT
+        ) {
+            Pair(GeometryHelper.getPoint(fromGeometry), GeometryHelper.getPoint(toGeometry))
+        } else if (fromGeometry.type() == FeatureType.POINT &&
+            toGeometry.type() == FeatureType.POLYGON
+        ) {
+            Pair(
+                GeometryHelper.getPoint(fromGeometry),
+                getPolygonCloserPoint(
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/uicomponents/map/geometry/line/MapLineRelationshipToFeature.kt:59]: I miss the unit test of this part. To make it simpler I would create another class to compute this part

File: app/src/main/java/org/dhis2/uicomponents/map/views/CarouselView.kt [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        this.carouselAdapter = adapter
+    }
+
+    fun attachToMapManager(mapManager: MapManager, callback: () -> Boolean) {
+        addOnScrollListener(object : OnScrollListener() {
+            override fun onScrollStateChanged(recyclerView: RecyclerView, newState: Int) {
+                super.onScrollStateChanged(recyclerView, newState)
+                if (newState == SCROLL_STATE_IDLE) {
+                    val feature = selectedFeature ?: mapManager.findFeatureFor(currentItem())
+//                    mapManager.mapLayerManager.selectFeature(feature)
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/uicomponents/map/views/CarouselView.kt:40]: can we remove this comment?

**Comment by JaimeToca** [comment]
Great job :1st_place_medal: . Add a couple of comments

File: app/src/main/java/org/dhis2/uicomponents/map/geometry/line/MapLineRelationshipToFeature.kt (hunk 2 above)
Comment (Balcan) [reply app/src/main/java/org/dhis2/uicomponents/map/geometry/line/MapLineRelationshipToFeature.kt:59]: done

File: app/src/main/java/org/dhis2/uicomponents/map/views/CarouselView.kt (hunk 3 above)
Comment (Balcan) [reply app/src/main/java/org/dhis2/uicomponents/map/views/CarouselView.kt:40]: done

**Pull request merged**